Sources/TuistAutomation/Device/PhysicalDevice.swift

```diff
@@ -15,13 +28,33 @@ public struct PhysicalDevice: Codable, Equatable, Identifiable {
             id: String = "id",
             name: String = "My iPhone",
             platform: Platform = .iOS,
-            osVersion: String = "17.4.1"
+            osVersion: String? = "17.4.1",
+            transportType: TransportType = .wifi,
+            connectionState: ConnectionState = .connected
         ) -> Self {
             .init(
                 id: id,
                 name: name,
                 platform: platform,
-                osVersion: osVersion
+                osVersion: osVersion,
+                transportType: transportType,
+                connectionState: connectionState
+            )
+        }
+
+        public func modified(
+            name: String? = nil,
+            osVersion: String? = nil,
+            transportType: TransportType? = nil,
+            connectionState: ConnectionState? = nil
+        ) -> PhysicalDevice {
+            .init(
+                id: id,
+                name: name ?? self.name,
+                platform: platform,
+                osVersion: osVersion ?? self.osVersion,
+                transportType: transportType ?? self.transportType,
+                connectionState: connectionState ?? self.connectionState
             )
```

Comment:
nit: I think I would rather end up with a bit of duplication here and there rather than adding this extra method